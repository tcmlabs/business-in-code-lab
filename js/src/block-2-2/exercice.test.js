const UserService = require("../userService");

async function dropUsers() {
  await UserService.client.query("DELETE FROM users;");
}

async function prepareUsers() {
  await UserService.client.query(
    `
    INSERT INTO users (
        id, first_name, last_name, created_at, updated_at
        )
    VALUES (1, 'Alice', 'Doe', '2020-03-04T14:12:49.424Z', '2020-03-04T14:12:49.424Z');
    
    INSERT INTO users (
        id, first_name, last_name, created_at, updated_at
    ) VALUES
    (2, 'Bob', 'Smith', '2020-03-04T14:12:49.424Z', '2020-03-04T14:12:49.424Z');
    `
  );
}

describe("user fetching", () => {
  beforeEach(async () => {
    await dropUsers();
    await prepareUsers();
  });

  afterEach(async () => {
    await dropUsers();
  });

  it("should fetch all users in a database", async () => {
    const users = await UserService.selectUsers();

    expect(users).toMatchSnapshot();
  });

  it("should fetch a user with a given id", async () => {
    const user = await UserService.selectUsersById(1);

    expect(user).toMatchSnapshot();
  });

  it("should return a full name", () => {
    // 1. Preparation phase
    const user = {
      first_name: "Alice",
      last_name: "Doe"
    };

    // 2. Test phase
    const fullName = UserService.fullName(user).fullName;

    // 3. Expectation phase
    const expected = "Alice Doe";
    expect(fullName).toBe(expected);
  });
});
