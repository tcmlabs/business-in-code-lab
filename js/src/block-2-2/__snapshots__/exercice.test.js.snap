// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user fetching should fetch a user with a given id 1`] = `
Object {
  "created_at": 2020-03-04T14:12:49.424Z,
  "first_name": "Alice",
  "fullName": "Alice Doe",
  "id": 1,
  "last_name": "Doe",
  "updated_at": 2020-03-04T14:12:49.424Z,
}
`;

exports[`user fetching should fetch all users in a database 1`] = `
Array [
  Object {
    "created_at": 2020-03-04T14:12:49.424Z,
    "first_name": "Alice",
    "id": 1,
    "last_name": "Doe",
    "updated_at": 2020-03-04T14:12:49.424Z,
  },
  Object {
    "created_at": 2020-03-04T14:12:49.424Z,
    "first_name": "Bob",
    "id": 2,
    "last_name": "Smith",
    "updated_at": 2020-03-04T14:12:49.424Z,
  },
]
`;

exports[`user fetching should fetch all users with full name in a database 1`] = `
Array [
  Object {
    "created_at": 2020-03-04T14:12:49.424Z,
    "first_name": "Alice",
    "fullName": "Alice Doe",
    "id": 1,
    "last_name": "Doe",
    "updated_at": 2020-03-04T14:12:49.424Z,
  },
  Object {
    "created_at": 2020-03-04T14:12:49.424Z,
    "first_name": "Bob",
    "fullName": "Bob Smith",
    "id": 2,
    "last_name": "Smith",
    "updated_at": 2020-03-04T14:12:49.424Z,
  },
]
`;
